rotkehlchen/chain/ethereum/modules/compound/v2/compound.py

```diff
@@ -103,21 +108,139 @@ def _get_apy(self, address: ChecksumEvmAddress, supply: bool) -> FVal | None:
         return apy
 
+    def _get_v3_interest_rates(
+            self,
+            tokens_and_types: dict['ChecksumEvmAddress', list[BalanceType]],
+    ) -> defaultdict['ChecksumEvmAddress', defaultdict[BalanceType, FVal]]:
+        """Make multicalls to the compound v3 tokens to get the lending and borrowing interest
+        rates from the contracts. First get the utilization for each token and then pass it in the
+        supply/borrow rate calls. Returns them as a dict of asset -> balance type -> rate."""
+        rates: defaultdict['ChecksumEvmAddress', defaultdict[BalanceType, FVal]] = defaultdict(lambda: defaultdict(FVal))  # noqa: E501
+        token_contract = EvmContract(
+            address=ZERO_ADDRESS,  # not used here
+            abi=self.ethereum.contracts.abi('COMPOUND_V3_TOKEN'),
+            deployed_block=0,  # not used here
+        )
+        calls = [  # prepare calls for utilizations of tokens
+            (token_address, token_contract.encode(method_name='getUtilization'))
+            for token_address in tokens_and_types
+        ]
+
+        try:
+            call_output = self.ethereum.multicall(calls=calls)  # get utilizations
+        except RemoteError as e:
+            log.error(f'Failed to query Compound v3 token utilizations while fetching the interest rates due to {e!s}')  # noqa: E501
+            return rates
+
+        rates_calls: dict[BalanceType, list[tuple['ChecksumEvmAddress', str]]] = defaultdict(list)
+        rates_calls_arguments: dict[BalanceType, list[str]] = defaultdict(list)
+        for idx, (token_address, _) in enumerate(calls):  # prepare calls for interest rates
+            for balance_type in tokens_and_types[token_address]:
+                decoded_utilization = token_contract.decode(
+                    result=call_output[idx],
+                    method_name='getUtilization',
+                )[0]
+                rates_calls[balance_type].append((
+                    token_address,
+                    token_contract.encode(
+                        method_name=f'get{"Supply" if balance_type == BalanceType.ASSET else "Borrow"}Rate',  # noqa: E501
+                        arguments=[decoded_utilization],  # passing decoded utilization
+                    ),
+                ))
+                rates_calls_arguments[balance_type].append(decoded_utilization)
+
+        try:
+            call_output = self.ethereum.multicall(  # get n supply and m borrow rates
+                calls=rates_calls[BalanceType.ASSET] + rates_calls[BalanceType.LIABILITY],  # n + m
+            )
+        except RemoteError as e:
+            log.error(f'Failed to query Compound v3 Supply and Borrow rates due to {e!s}')
+            return rates
+
+        for idx, (token_address, _) in enumerate(rates_calls[BalanceType.ASSET]):
+            rates[token_address][BalanceType.ASSET] = (FVal(token_contract.decode(
```

Comment:
And this can raise `DeserializationError` in the case the contract returns unknown/unreadable number

```suggestion
            rates[token_address][BalanceType.ASSET] = (deserialize_fval(token_contract.decode(
```